### PR TITLE
feat(message-coder): add `byteArray` & related coders

### PR DESCRIPTION
**Pull request by eventualbuddha**
## Overview
These coder handle byte arrays, especially the common pattern of an encoded length of bytes followed by the number of bytes specified by that length.

## Demo Video or Screenshot
```ts
export const RealTimeExchangeResponse = message({
  startOfPacket: literal(0x82),
  requestId: uint8(),
  token: literal(TOKEN),
  returnCode: uint8(),
  optionalBytes: byteArrayWithLengthPrefix(uint8()),
});
```

## Testing Plan
Automated.

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by kshen0** [comment]
> I don't love the name dynamicBytes. Got a better suggestion?
`variableLengthByteSequence`? `arbitraryBytes`?

`unboundedBytes` would be consistent with `unboundedString` but I guess in this case it is actually bounded

File: libs/message-coder/src/dynamic_bytes.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  expect(coder.default()).toEqual(Uint8Array.of());
+});
+
+test('bitLength', () => {
+  const coder = dynamicBytes(uint8());
+  expect(coder.bitLength(Uint8Array.of(1, 2, 3))).toEqual(ok(8 + 24));
+});
+
+test('encode', () => {
+  const coder = dynamicBytes(uint8());
```
Comment (kshen0) [inline libs/message-coder/src/dynamic_bytes.test.ts:24]: It's overkill to test every coder, but can we test `dynamicBytes(literal(...))`? `literal` and `uint8` are the 2 cases I've seen in the paper-handler manual.

File: libs/message-coder/src/dynamic_bytes.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ * @example
+ *
+ * ```ts
+ * import { dynamicBytes, uint8 } from '@votingworks/message-coder';
+ *
+ * const coder = dynamicBytes(uint8());
+ * coder.encode(Uint8Array.from([1, 2, 3]));
+ * // => Uint8Array.of(3, 1, 2, 3)
+ * //           length ┘  └──┴──┴ data
```
Comment (kshen0) [inline libs/message-coder/src/dynamic_bytes.ts:98]: Very useful example

**Review by kshen0 — APPROVED** [review]

File: libs/message-coder/src/dynamic_bytes.test.ts (hunk 1 above)
Comment (eventualbuddha) [reply libs/message-coder/src/dynamic_bytes.test.ts]: `literal` doesn't actually work here, but I added a couple variants to achieve what you want:

```ts
byteArray(64); // encodes/decodes 64 bytes as-is
byteArrayWithLengthPrefix(uint8()); // encodes/decodes 0-255 bytes by encoding the length followed by the bytes
byteArrayWithLengthPrefix(64); // encodes/decodes 64 bytes by encoding the (fixed) length followed by the bytes
```

**Review by kshen0 — APPROVED** [review]
Name change LGTM. Thanks for adding the fixed length decoder.

**Pull request merged**